src/components/bar-chart/bar-chart.js

```diff
@@ -11,15 +13,53 @@ export class BarChartComponent {
     this.barChartCtx = this.mountPoint.querySelector(".chart__visual");
   }
 
-  createBarChart() {
+  updateChart(data) {
+    let amount = data.amount;
+    if (data.account.currency != "UAH") {
+      amount = CurrencyConverterUAH.convert(data.account.currency, amount);
+    }
+    data.type == "-"
+      ? (this.dataset.expence += amount)
+      : (this.dataset.income += amount);
+    this.drawChangedChart();
+  }
+
+  drawChangedChart() {
+    if (!this.barChart) this.drawChart();
+    this.barChart.data.datasets[0].data = [
+      this.dataset.income,
+      this.dataset.expence
+    ];
+    this.barChart.update();
+  }
+
+  createFromList(list) {
+    this.dataset = list.reduce(
+      (data, item) => {
+        let amount = item.amount;
+        if (item.account.currency != "UAH") {
+          amount = CurrencyConverterUAH.convert(item.account.currency, amount);
+        }
+        item.type == "-" ? (data.expence += amount) : (data.income += amount);
```

Comment:
do not mutate paramters -- e.g `item`, `data`

Reply:
data is my current value for reduce method